### PR TITLE
Config file 

### PR DESCRIPTION
**Pull request by amao9098**
* Create json config files for azure, detail, and default situation
* Integrate json config files to command generation 
* Refer to Toolname.Config.md for more information on each config file

### CODE REVIEW

File: ntttcp/ntttcp.Config.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{
+    "NtttcpDefault": {
+        "Iterations"  : 5,
+        "StartPort"   : 50002,
+        "tcp" : {
+            "BufferLen"   : [65536],
```
[the remark this answers is not in the document]
Comment (amao9098) [reply ntttcp/ntttcp.Config.json:6]: This is the default for ntttcp. Windows uses 65536 as well.

File: ntttcp/ntttcp.Config.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    "NtttcpDetail": {
+        "Iterations"  : 6,
+        "StartPort"   : 50002,
+        "tcp" : {
+            "BufferLen"   : [65536],
+            "Connections" : [1, 2, 4, 8, 16, 32, 64, 128, 256, 512],
+            "Options"     : ""
+        },
+        "udp" : {
+            "BufferLen"   : [1450],
```
[the remark this answers is not in the document]
Comment (amao9098) [reply ntttcp/ntttcp.Config.json:51]: That was the highest BuffLen I got with udp without running into fragmentation when I was testing previous.

File: ntttcp/ntttcp.Config.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    "NtttcpAzure": {
+        "Iterations"  : 6,
+        "StartPort"   : 50002,
+        "tcp" : {
+            "BufferLen"   : [65536],
+            "Connections" : [1, 64, 256],
+            "Options"     : ""
+        },
+        "udp" : {
+            "BufferLen"   : [1450],
```
[the remark this answers is not in the document]
Comment (amao9098) [reply ntttcp/ntttcp.Config.json:31]: No it doesn't, windows uses 1472, but I can align.

File: ntttcp/ntttcp.Config.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    "NtttcpAzure": {
+        "Iterations"  : 6,
+        "StartPort"   : 50002,
+        "tcp" : {
+            "BufferLen"   : [65536],
+            "Connections" : [1, 64, 256],
+            "Options"     : ""
+        },
+        "udp" : {
+            "BufferLen"   : [1472],
```
Comment (vidyas00) [inline ntttcp/ntttcp.Config.json:31]: we might want to add a testcase for bufferlen 1372 to validate throughtput without getting into the state of fragmentation /hash collisions.. Ok to fast follow

File: ntttcp/ntttcp.Config.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        },
+        "Warmup"      : 2,
+        "Cooldown"    : 2,
+        "Runtime"        : 60,
+        "RecvOptions" : "",
+        "SendOptions" : ""
+    },
+
+    "NtttcpDetail": {
+        "Iterations"  : 6,
```
Comment (vidyas00) [inline ntttcp/ntttcp.Config.json:43]: Might be way too much. Should we stick to the same # of iterations as the windows NPT runs? 
Detail runs are typically quick sampling tests. so the goal here is more combinations and fewer iterations. Runtime could be small too

File: ntttcp/ntttcp.Config.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            "Options"     : ""
+        },
+        "udp" : {
+            "BufferLen"   : [512, 1372, 1472],
+            "Connections" : [1, 2, 4, 8, 16, 32, 64, 128, 256, 512],
+            "Options"     : ""
+        },
+        "Warmup"      : 2,
+        "Cooldown"    : 2,
+        "Runtime"        : 60,
```
Comment (vidyas00) [inline ntttcp/ntttcp.Config.json:57]: for detail, I'd recommend 10 seconds (same as windows)

File: ntttcp/ntttcp.TESTGEN.ps1 [hunk 7]
```diff
@@ -123,104 +113,95 @@ function test_ntttcp {
...
-    if (Test-Path $g_ConfigFile) {
-        test_ntttcp -OutDir $dir -ConfigFile $g_ConfigFile -SendDir $sendDir -RecvDir $recvDir
-    } else {
+    try {
+        input_display
+        $allConfig = Get-Content -Path "$PSScriptRoot/ntttcp.Config.json" | ConvertFrom-Json
+        # Get config variables
+        [Object] $g_Config = $allConfig.("Ntttcp$Config")
+        if ($null -eq $g_Config) {
+            Write-Host "Ntttcp$Config does not exist in ./ntttcp/ntttcp.Config.json. Please provide a valid config"
```
Comment (vidyas00) [inline ntttcp/ntttcp.TESTGEN.ps1:178]: $PSScriptRoot/ntttcp.Config.json

**Review by vidyas00 — APPROVED** [review]

**Comment by vidyas00** [comment]
Approved with suggestions.

**Pull request merged**